Vizzuality/widget-editor
# Implement the end-user filters feature

This PR implements the end-user filters feature required by the FA contract.

<p align="center">
<img width="1420" alt="Bar chart with two end user filters and their settings on the right" src="https://user-images.githubusercontent.com/6073968/90551559-6afdab80-e189-11ea-97ba-786a8f23ffa1.png">
</p>

Minimal styles have been given to the filters so they can be modified in the host app easily.

By default, the feature is disabled.

## Testing instructions

1. Instantiate this dataset: `852f2275-91a8-4500-9f10-89880dc53f22`
2. Create a bar chart
3. Add a new end-user filter based on the “category” column

Below the visualisation, a “category” filter must appear.

4. Change the value of the filter

The visualisation must be updated accordingly. You can use the “category” field for the X axis to check if the visualisation is updated correctly.

5. Select a 3rd dimension for the chart, like “string”
6. Change the value of the filter

Make sure that the colour of the bars and the legend are always in sync.

7. Switch to the pie chart
8. Play with the filter again

Also make sure the colour of the slices are always in sync with the legend.
Please note that when combining end-user filters and the pie or donut chart, the slice count setting is ignored and the legend displays all of the dataset's values (not just the ones of the slices on screen).

9. Open the chart specific settings

Make sure a note tells the user that the slice count settings are ignored

10. Click the playground's button to get the editor state

Make sure the `paramsConfig` object contains a `endUserFilters` array with the list of columns your have selected in the UI.

## Pivotal Tracker

[Task](https://www.pivotaltracker.com/story/show/171081209).
